### PR TITLE
Create evals for system jobs on node registration

### PR DESCRIPTION
**Pull request by dadgar**
This PR adds:
* A secondary index to allow job indexing by scheduler type.
* Creates evals for system jobs when a node transitions from Init to Ready.

### CODE REVIEW

File: nomad/node_endpoint.go [hunk 1]
```diff
@@ -479,6 +486,37 @@ func (n *Node) createNodeEvals(nodeID string, nodeIndex uint64) ([]string, uint6
 		evalIDs = append(evalIDs, eval.ID)
 	}
 
+	var sysJobs []*structs.Job
+	if nextJob != nil {
+		sysJobs = append(sysJobs, nextJob.(*structs.Job))
+		for job := sysJobsIter.Next(); job != nil; job = sysJobsIter.Next() {
+			sysJobs = append(sysJobs, job.(*structs.Job))
+		}
+	}
```
Comment (ryanuber) [inline nomad/node_endpoint.go]: Maybe just move this above the fast-path so you don't have to save the first job or duplicate the append().

File: nomad/state/schema.go [hunk 2]
```diff
@@ -91,6 +91,15 @@ func jobTableSchema() *memdb.TableSchema {
 					Lowercase: true,
 				},
 			},
+			"type": &memdb.IndexSchema{
+				Name:         "type",
+				AllowMissing: false,
+				Unique:       false,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Type",
+					Lowercase: false,
```
Comment (ryanuber) [inline nomad/state/schema.go]: Scheduler types probably should be case-insensitive, even though its mostly hard-coded everywhere we probably shouldn't open this up. Is there a reason we would want this to be case-sensitive?

**Comment by ryanuber** [comment]
@dadgar this LGTM, just minor comments/questions!

File: nomad/state/schema.go (hunk 2 above)
Comment (dadgar) [inline nomad/state/schema.go]: The reason I didn't is that when you look-up a scheduler it is just looking up in a map where the keys are lower-case: https://github.com/hashicorp/nomad/blob/master/scheduler/scheduler.go#L23

So if you are not lower-case you wouldn't even be able to pick a scheduler. I think the correct fix is to lower-case user input when parsing the job file.

File: nomad/node_endpoint.go (hunk 1 above)
Comment (dadgar) [inline nomad/node_endpoint.go]: Yeah it might have been an over-optimization. Will do

File: nomad/node_endpoint.go [hunk 3]
```diff
@@ -154,7 +154,8 @@ func (n *Node) UpdateStatus(args *structs.NodeUpdateStatusRequest, reply *struct
 	}
 
 	// Check if we should trigger evaluations
-	if structs.ShouldDrainNode(args.Status) {
+	initToReady := node.Status == structs.NodeStatusInit && args.Status == structs.NodeStatusReady
```
Comment (armon) [inline nomad/node_endpoint.go]: It should also trigger if we go from a Terminal state to Ready. e.g. Node is ready, then goes down (system scheduler stops the allocs), node comes back and goes ready (system scheduler must create new allocs)

Comment (dadgar) [inline nomad/node_endpoint.go]: Done

**Pull request merged**